apps/web-connect/src/features/checkout/components/crossmint/CrossmintModal.tsx

```diff
@@ -15,11 +14,36 @@ interface CrossmintModalProps {
     onClose: () => void;
 }
 
-
-const CrossmintModal: React.FC<CrossmintModalProps> = ({ isOpen, onClose, totalPriceInUsdc, totalQty, promoCode }) => {
+const CrossmintModal: React.FC<CrossmintModalProps> = ({ isOpen, onClose, totalQty, totalPriceInUsdc, promoCode }) => {
     const collectionId = config.crossmintCollectionId;
-    const clientApiKey = config.crossmintClientApiKey;
     const { address } = useAccount();
+    const { order } = useCrossmintCheckout();
+    const { setMintWithCrossmint } = useWebBuyKeysContext();
+    const [mintTxData, setMintTxData] = useState<MintWithCrossmintStatus>({ txHash: "", orderIdentifier: "" });
+    
+    const handleClose = () => {
+        if (mintTxData.txHash === "") {
+            setMintWithCrossmint({ txHash: "", orderIdentifier: "" });
+        } else {
+            setMintWithCrossmint(mintTxData);
```

Comment:
`setMintWithCrossmint(mintTxData.txHash === "" ? { txHash: "", orderIdentifier: "" } : mintTxData);` is how i would maybe do it but def optional. just an idea

Reply:
Done, great suggestion!